MapboxMobileEvents/MMEEvent.m

```diff
@@ -12,6 +12,17 @@ + (instancetype)turnstileEventWithAttributes:(NSDictionary *)attributes {
     return turnstileEvent;
 }
 
++ (instancetype)telemetryMetricsEventWithDateString:(NSString *)dateString attributes:(NSDictionary *)attributes {
+    MMEEvent *telemetryMetrics = [[MMEEvent alloc] init];
```

Comment:
Would having an additional `MMEEvent` initializer be a breaking change?

Reply:
We should be able do that. Have a convenience initializer that takes in the `name`, `dateString`, and additional `attributes`. Later we can look at getting the `dateString` to be an actual `date` that will be easier to manipulate.

```
MMEEvent *telemetryMetrics = [[MMEEvent alloc] initWithName:name created:dateString attributes:attributes];
```

Reply:
Perhaps we hold off on this for when we consider throwing around `NSDate`s? Otherwise we might end up having to remove an initializer later. Up to you :)

Reply:
Well, either way. The initializer should take the date as a formatted string to match our schema. Any magic would have to happen before we initialize the event in this case -- which is fine but will probably be a major version change at that point since we'll require an NSDate as a parameter for this method. 

The bad thing about this change is that it would put more pressure on the MME library to keep the date format matching the schema. Which is something we were trying to move away from.

Reply:
Would it make sense to pass in an `NSDateFormatter` as a parameter? Then whatever desired format could be used when the event is generated.

Reply:
Idle thought - should the date format be supplied by the configuration api?

Reply:
Possibly... but if there ever came a time where an event required a separate date format for whatever reason then we may need to setup an additional configuration for specific events. 

IMO adding a `NSDateFormatter` parameter to our `MMEEvent` constructors makes the most sense for our current setup. This would only require a small change for our clients and would provide us with more flexibility to adjust any dates as necessary. Still, I think this would be considered a major release and we should hold off until we're ready for that. We could look at building out a more complex system to provide common type definitions through a configuration at a later time if we can find more benefit in doing so.

Reply:
👍